packages/wrangler/templates/middleware/middleware-ensure-req-body-drained.ts

```diff
@@ -0,0 +1,14 @@
+import type { Middleware } from "./common";
+
+const drainBody: Middleware = async (request, env, _ctx, middlewareCtx) => {
+	try {
+		return await middlewareCtx.next(request, env);
+	} finally {
+		if (request.body !== null && !request.bodyUsed) {
+			const reader = request.body.getReader();
+			while (!(await reader.read()).done) {}
```

Comment:
Why was the change from cancelling the stream required?

Reply:
Cancelling the stream, via `request.body.cancel()` did not resolve the problem.
The original PR drained the stream by dumping it into a buffer (e.g. `request.arrayBuffer()` but this creates unnecessary memory churn.
So the current approach literally just iterates over the chunks and throws them away.